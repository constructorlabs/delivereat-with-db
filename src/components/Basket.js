import React from "react";

import "../styles/Basket.scss";
import cx from "classnames";

class Basket extends React.Component {
  constructor() {
    super();
    this.state = { order: {}};
  }

  

  render() {
    console.log(this.props.basket);

    const basketView = cx("hidden", { "basket": this.props.showBasket });
    console.log(this.props.showBasket);
    return (
      
        <div className={basketView}>
          <h4 className="BasketCloser" onClick={() => this.props.closeBasket()}>
            X
          </h4>
          <ul className="basket__contents">
            {this.props.basket.map(item => {
              return (
                <li key={item.menuItemId}>
                  <div className="basket__row">
                    <h2>{this.props.menu[item.menuItemId].name}</h2>
                    <div className="basket__order">
                      <button
                        className="orderbutton__plus"
                        onClick={() =>
                          this.props.basketReceiveAdd(item.menuItemId)
                        }
                      >
                        +
                      </button>

                      <h3>{item.quantity}</h3>
                      <button
                        className="orderbutton__minus"
                        onClick={() =>
                          this.props.basketReceiveRemove(item.menuItemId)
                        }
                      >
                        -
                      </button>
                    </div>
                    <h2 className="basket__price">
                      £{this.props.menu[item.menuItemId].price * item.quantity}
                    </h2>
                  </div>
                </li>
              );
            })}
          </ul>
          <span>
            <h1>Delivery Charge</h1>
            <h1>£2.5</h1>
          </span>
          <span>
            <h1>Order Total</h1>
            {/* <h1>{this.props.calculate}</h1> */}
            {/* <h1>{this.props.menu[item.menuItemId].price * item.quantity}</h1> */}
          </span>
          <button onClick={() => this.props.receiveOrder(this.props.basket)}>
            Place Order
          </button>
        </div>
      
    );
  }
}

export default Basket;
