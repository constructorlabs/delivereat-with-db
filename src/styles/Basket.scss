@import "defaults";

.basket {

  top:0;
  animation: bounce 1s;

  &__order {
    display: flex;
    flex-direction: row;
    // justify-content: space-around;
    align-items: baseline;
    // display: inline;
  }
}

.hidden {
  position: fixed;

 
  bottom: -100%;
  left: 0px;
  width: 100vw;
  height: 100vh;
  background-color: aquamarine;
  transition: bottom 0.5s;
  margin-right: 0px;
  opacity: 0.9;
  z-index: 1;
  animation: bounce 1s;
}
