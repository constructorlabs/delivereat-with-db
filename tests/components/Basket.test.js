import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Basket from '../../src/components/Basket';

const total = 12;
const stage = 'menu';
const menu = [
  {
    id: 1,
    name: 'Espresso',
    description: 'Single shot, 30ml',
    price: '2.00',
  },
  {
    id: 2,
    name: 'Ristretto',
    description: 'Concentrated espresso, 22ml',
    price: '2.00',
  },
  {
    id: 3,
    name: 'Macchiato',
    description: '60ml espresso + a dab of milk foam',
    price: '2.20',
  },
];
const mockChangeStage = jest.fn();
const order = { id: 10, items: [[1, 5]] };
const mockAddToOrder = jest.fn();
const mockRemoveFromOrder = jest.fn();
const mockGetInput = jest.fn();
const mockCheckout = jest.fn();
const readyToCheckout = false;

describe('Basket', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Basket
        total={total}
        stage={stage}
        menu={menu}
        changeStage={mockChangeStage}
        order={order}
        addToOrder={mockAddToOrder}
        removeFromOrder={mockRemoveFromOrder}
        checkout={mockCheckout}
        getInput={mockGetInput}
        readyToCheckout={readyToCheckout}
      />,
    );
  });
  afterEach(() => {
    wrapper = null;
  });
  test('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Basket
          total={total}
          stage={stage}
          menu={menu}
          changeStage={mockChangeStage}
          order={order}
          addToOrder={mockAddToOrder}
          removeFromOrder={mockRemoveFromOrder}
          checkout={mockCheckout}
          getInput={mockGetInput}
          readyToCheckout={readyToCheckout}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('click changes stage', () => {
    wrapper.find('.basket-details__close').simulate('click');
    expect(mockChangeStage.mock.calls).toEqual([['menu']]);
  });
  test('enter phone number', () => {
    wrapper
      .find('.basket-details__phone-input')
      .simulate('change', { target: { value: '7777777777' } });
    expect(mockGetInput.mock.calls).toEqual([['7777777777']]);
  });
  test('checkout', () => {
    wrapper.find('.basket-checkout').simulate('click');
    expect(mockCheckout).toHaveBeenCalled();
  });
  test('free delivery over £10', () => {
    const deliveryCharge = wrapper.find('.basket-details__delivery-price').text();
    expect(deliveryCharge).toEqual('£0.00');
  });
});
