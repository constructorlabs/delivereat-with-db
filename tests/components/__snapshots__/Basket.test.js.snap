// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket matches the snapshot 1`] = `
<div
  className="basket"
>
  <div
    className="basket-details"
  >
    <button
      className="basket-details__close"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
    <div
      className="basket-item"
    >
      <div
        className="basket-item__title"
      >
        <div
          className="basket-item__name"
        >
          Espresso
        </div>
        <div
          className="basket-item__buttons"
        >
          <button
            className="basket-item__remove-button"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-minus-circle"
            />
          </button>
          <div
            className="basket-item__count"
          >
            5
          </div>
          <button
            className="basket-item__add-button"
            onClick={[Function]}
            type="button"
          >
            <i
              className="fas fa-plus-circle"
            />
          </button>
          <div
            className="basket-item__line-total"
          >
            £10.00
          </div>
        </div>
      </div>
    </div>
    <div
      className="basket-details__delivery"
    >
      <div
        className="basket-details__delivery-text"
      >
        Free delivery
      </div>
      <div
        className="basket-details__delivery-price"
      >
        £0.00
      </div>
    </div>
    <div
      className="basket-details__total"
    >
      <div
        className="basket-details__total-text"
      >
        Order total
      </div>
      <div
        className="basket-details__total-price"
      >
        £12.00
      </div>
    </div>
    <div
      className="basket-details__phone"
    >
      <div
        className="basket-details__phone-text"
      >
        Enter phone number:
      </div>
      <div
        className="basket-details__phone-number"
      >
        <div
          className="basket-details__phone-country"
        >
          +44
        </div>
        <input
          className="basket-details__phone-input"
          onChange={[Function]}
          placeholder="Phone number"
          type="text"
        />
      </div>
    </div>
  </div>
  <button
    className="basket-summary basket-summary--visible"
    onClick={[Function]}
    type="button"
  >
    <div
      className="basket-summary__count"
    >
      5
    </div>
    <div
      className="basket-summary__text"
    >
      View Basket
    </div>
    <div
      className="basket-summary__total"
    >
      £12.00
    </div>
  </button>
  <button
    className="basket-checkout"
    onClick={[Function]}
    type="button"
  >
    Checkout
  </button>
</div>
`;
